### PR TITLE
fix(auth): add 401 response to member invite

### PR DESCRIPTION
**Pull request by mifu67**
In the event that a user is logged into Sentry with an account that is different than the account invited to join an organization, we wish to prompt them to log out and try again. To do this, we introduce a 401 UNAUTHORIZED response if the requesting user's email does not match the email in the invite context.

This does not result in any changes on the user side, because the same warning message is rendered for all errors. Frontend changes will come in a subsequent PR.

### CODE REVIEW

File: src/sentry/api/endpoints/accept_organization_invite.py [hunk 1]
```diff
@@ -110,6 +110,13 @@ class AcceptOrganizationInvite(Endpoint):
     def respond_invalid() -> Response:
         return Response(status=status.HTTP_400_BAD_REQUEST, data={"details": "Invalid invite code"})
 
+    @staticmethod
+    def respond_unauthorized() -> Response:
```
Comment (cathteng) [inline src/sentry/api/endpoints/accept_organization_invite.py:114]: does responding with 401 force re-login?

Comment (cathteng) [reply src/sentry/api/endpoints/accept_organization_invite.py:114]: from memory this makes some error message appear but does not force somebody to re-auth

Comment (mifu67) [reply src/sentry/api/endpoints/accept_organization_invite.py:114]: That's the goal—show a screen that lets the user know that they're signed into X account, then include a link for them to log out and be redirected back to the invite page.

Comment (mifu67) [reply src/sentry/api/endpoints/accept_organization_invite.py]: We include a link for the user to log out on the frontend.

Comment (cathteng) [reply src/sentry/api/endpoints/accept_organization_invite.py]: is this something we need to account for before we make this change?

Comment (cathteng) [reply src/sentry/api/endpoints/accept_organization_invite.py]: a video of the flow would help

Comment (mifu67) [reply src/sentry/api/endpoints/accept_organization_invite.py]: This is implemented in the frontend, so no changes for now. Currently, the frontend displays the same error view for all errors.

**Review by cathteng — COMMENTED** [review]
chatted in slack, but i think we should make sure we are properly fetching the invite context for the org member being invited before adding the new 401 response

File: src/sentry/api/endpoints/accept_organization_invite.py [hunk 2]
```diff
@@ -136,7 +137,7 @@ def get(
         invite_context = get_invite_state(
             member_id=int(member_id),
             organization_id_or_slug=organization_id_or_slug,
-            user_id=request.user.id,
+            user_id=None,  # NOTE (mifu67): we want to get the invite context using the member ID only
```
Comment (cathteng) [inline src/sentry/api/endpoints/accept_organization_invite.py:140]: nit: you can also delete this, it defaults to `None`

File: src/sentry/api/endpoints/accept_organization_invite.py [hunk 3]
```diff
@@ -137,6 +148,15 @@ def get(
...
         organization = invite_context.organization
 
+        # NOTE (mifu67): we get the invite context without passing in the user_id, so the
+        # invite context is generated solely using the passed member_id. Then, compare the
+        # email of the current session user against the invite context member email. If
+        # they're different, raise a 401 Unauthorized error, which causes the frontend to
+        # render a view prompting the user to log out and try again.
+        if not isinstance(request.user, AnonymousUser):
+            if organization_member.email != request.user.email:
+                return self.respond_unauthorized(request.user.email)
```
Comment (cathteng) [inline src/sentry/api/endpoints/accept_organization_invite.py:158]: can we test returning this 401?

**Review by cathteng — COMMENTED** [review]
very elegant. test pls

**Pull request closed without merging**